lib/mongoid/fields/standard.rb

```diff
@@ -231,7 +231,11 @@ def included?(fields)
       # @since 2.4.4
       def evaluated_default(doc)
         if default_val.respond_to?(:call)
-          evaluate_default_proc(doc)
+          begin
+            evaluate_default_proc(doc)
+          rescue NoMethodError => e
+            return nil
+          end
```

Comment:
We actually need to redesign a little so that methods referencing an association aren't called until after the relation is set on the new object. The initialize method of Document determines the order in which this setup happens but changing the order wouldn't be enough-- we need to change fields.rb to account for a dependency of a certain field on an association.